plan/plan_to_pb.go

```diff
@@ -120,12 +120,9 @@ func (p *PhysicalIndexScan) ToPB(ctx sessionctx.Context) (*tipb.Executor, error)
 	tableColumns := p.Table.Cols()
 	for _, col := range p.schema.Columns {
 		if col.ID == model.ExtraHandleID {
-			columns = append(columns, &model.ColumnInfo{
-				ID:   model.ExtraHandleID,
-				Name: model.ExtraHandleName,
-			})
+			columns = append(columns, model.NewExtraHandleColInfo())
 		} else {
-			columns = append(columns, tableColumns[col.Position])
+			columns = append(columns, model.FindColumnInfo(tableColumns, col.ColName.L))
```

Comment:
How about find column by ID?

Reply:
@coocood `IndexScan`'s `Column` in `Schema` is generated from `IndexInfo.IndexColumns` which don't have `ID`. So if we find column by ID here. We need to find its ID by name when create schema.